pkg/tools/serialize/serialize.go

```diff
@@ -24,15 +24,11 @@ import (
 	"sync/atomic"
 )
 
-const (
-	channelSize = 100
-)
-
 // Executor - a struct that can be used to guarantee exclusive, in order execution of functions.
 type Executor struct {
-	execCh chan func()
-	init   sync.Once
-	count  int32
+	mutex sync.Mutex
```

Comment:
Mutex's do not guarantee ordering:

https://golang.org/src/sync/mutex.go?s=765:813#L42

So it's entirely possible for events to be queued out of order due to the out of order acquisition of a mutex.  Use of a mutex here basically yields the ordering guarantee.

Reply:
Yes, you're correct and it is a real problem. It seems to me we need to create our own analog of channel or FIFO mutex. Any ideas?

Reply:
Also, we can just increase the capacity of the channel and add notes

Reply:
@edwarnicke  Also, we can just increase the capacity of the channel and add notes

Reply:
@edwarnicke Lets consider use cases

In what follows, by the **Executor**, we mean `seriazlie.Executor` based on the mutex.
 
Usecases:

## 1. One goroutine using the Executor.
In this case, tasks will be executed FIFO because mutex will be used consistently. 

## 2. Two or more goroutine using the Executor
In this case, tasks will be executed FIFO for each goroutine but the order of goroutine maybe is not FIFO. This thing is not guaranteeing even by channel.


@edwarnicke Can you provide an example of what are you meaning?